src/NumericalAlgorithms/Spectral/SwshTransform.hpp

```diff
@@ -782,5 +782,69 @@ using make_transform_list_from_derivative_tags =
         tmpl::transform<DerivativeTagList,
                         tmpl::bind<db::remove_tag_prefix, tmpl::_1>>,
         decltype(std::make_integer_sequence<int, 5>{})>::type;
+
+/// \ingroup SwshGroup
+/// \brief Convert spin-weighted spherical harmonic data to a new set of
+/// collocation points (either downsampling or upsampling)
+template <int Spin>
+void interpolate_to_collocation(
+    const gsl::not_null<SpinWeighted<ComplexDataVector, Spin>*> target,
+    const SpinWeighted<ComplexDataVector, Spin>& source,
+    const size_t target_l_max, const size_t source_l_max,
+    const size_t number_of_radial_points) noexcept {
+  auto source_modes =
+      swsh_transform(source_l_max, number_of_radial_points, source);
+  SpinWeighted<ComplexModalVector, Spin> target_modes{
+      size_of_libsharp_coefficient_vector(target_l_max) *
+      number_of_radial_points};
+
+  const auto& target_coefficient_metadata =
+      cached_coefficients_metadata(target_l_max);
+  for(size_t i = 0; i < number_of_radial_points; ++i) {
+    auto source_coefficient_metadata_iterator =
+        cached_coefficients_metadata(source_l_max).begin();
+    for (const auto& coefficient : target_coefficient_metadata) {
+      // note: this assumes the libsharp coefficient ordering for optimizations
+      while (source_coefficient_metadata_iterator !=
```

Comment:
can you add a comment stating what this sets `source_coefficient_metadata_iterator` to so we don't have to figure it out...